bootstrapping/bootstrapping.go

```diff
@@ -0,0 +1,428 @@
+		currentlyProcessableFiles = append(currentlyProcessableFiles, policyStatus.filename) // Downloaded tgz file names
+		currentlyProcessableFiles = append(currentlyProcessableFiles, policyStatus.name)     // Uncompressed folder names
+	}
+
+	// evaluates working folder
+	files, err := ioutil.ReadDir(directoryPath)
+	if err != nil {
+		// TODO should it be an error?
+		log.Warn("Cannot read directory: ", directoryPath, err)
+	}
+
+	// removes files not regarding to any of current policy files
+	for _, f := range files {
+		if !utils.Contains(currentlyProcessableFiles, f.Name()) {
+			log.Debug("Removing: ", f.Name())
+			if err := utils.RemoveFileInfo(f, strings.Join([]string{directoryPath, string(os.PathSeparator), f.Name()}, "")); err != nil {
+				// TODO should it be an error?
+				log.Warn("Cannot remove: ", f.Name(), err)
+			}
+		}
+	}
+	return nil // TODO should it be managed as error?
+}
+
+// saveAttributes stores the attributes as JSON in a file with name `attrs-<attribute_revision_id>.json`
+func saveAttributes(bsStatus *bootstrappingStatus) error {
+	log.Debug("saveAttributes")
+
+	attrs, err := json.Marshal(bsStatus.attributes.rawData)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(bsStatus.attributes.filePath, attrs, 0600); err != nil {
+		return err
+	}
+	return nil
+}
+
+
+//For every policy file, apply them doing the following:
+//	* Extract the tarball to a temporal work directory DIR
+//	* Run  `cd DIR; chef-client -z -j path/to/attrs-<attribute_revision_id>.json` while sending the stderr and stdout in bunches of 10 lines to the
+// platform via `POST /blueprint/bootstrap_logs` (this resource is a copy of POST /command_polling/bootstrap_logs used in the command_polling command).
+// If the command returns with a non-zero value, stop applying policy files and continue with the next step.
+
+// TODO On the first iteration that applies successfully all policy files (runs all `chef-client -z` commands obtaining 0 return codes) only, run the boot scripts for the server by executing the `scripts boot` sub-command (as an external process).
+// TODO Just a POC, an starging point. To be completed...
+func processPolicyFiles(bootstrappingSvc *blueprint.BootstrappingService, bsStatus *bootstrappingStatus) error {
+	log.Debug("processPolicyFiles")
+
+	// Run  `cd DIR; chef-client -z -j path/to/attrs-<attribute_revision_id>.json` while sending the stderr and stdout in bunches of
+	// 10 lines to the platform via `POST /blueprint/bootstrap_logs` (this resource is a copy of POST /command_polling/bootstrap_logs used in
+	// the command_polling command). If the command returns with a non-zero value, stop applying policyfiles and continue with the next step.
+	for _, policyStatus := range bsStatus.policiesStatus {
+		log.Warn(policyStatus.folderPath)
+
+		// TODO cd <policyStatus.folderPath>; chef-client -z -j <bsStatus.attributes.filePath>`
+		command := "ping -c 100 8.8.8.8"
```

Comment:
Just to "play" while code is modelling

Reply:
One thing we will have to introduce here is different commands depending on the os being windows or linux. The windows command will need an additional SET command run:
```
cd <PolicyStatus.folderPath DIR>
SET "PATH=%PATH%;C:\ruby\bin;C:\opscode\chef\bin;C:\opscode\chef\embedded\bin"
chef-client -z -j <path/to/attrs-<attribute_revision_id>.json>
```

Reply:
Well, currently it saves the received commands into a file and then it's invoked to run...
That's should be valid, right? What's you point of view?